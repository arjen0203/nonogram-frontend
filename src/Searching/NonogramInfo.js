import React, {Component} from 'react';

class NonogramInfo extends Component {
    render() {
        return (
            <div className="nonoogram-info" onClick={() => this.props.onClick()}>
                <div className="gram-info-name">{this.props.info.name}</div>
                <div className="gram-info-creater">By: {this.props.info.creater}</div>
            </div>
        );
    }
}

export default NonogramInfo;