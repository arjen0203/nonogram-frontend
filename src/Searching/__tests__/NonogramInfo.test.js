import React from "react";
import ReactDOM from 'react-dom';
import NonogramInfo from "../NonogramInfo";
import {render, cleanup} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect"

import renderer from "react-test-renderer";

const mockInfo = {id: 5, name: "puzzle", creater: "rens"};

afterEach(cleanup);

it('Renders without crashing', function () {
	const mockInfo = {id: 0, name: "", creater: ""};

	const div = document.createElement('div');
	ReactDOM.render(<NonogramInfo info={mockInfo}></NonogramInfo>, div);
});

it('Render correctly', function () {
	const mockInfo = {id: 5, name: "puzzle", creater: "rens"};

	const comp = render(<NonogramInfo info={mockInfo}></NonogramInfo>).getByTestId("nonoInfo");
	expect(comp).toHaveTextContent("puzzle");
	expect(comp).toHaveTextContent("By: rens");
});

it('Render correctly', function () {
	const mockInfo = {id: 5, name: "Crazy", creater: "user123"};

	const comp = render(<NonogramInfo info={mockInfo}></NonogramInfo>).getByTestId("nonoInfo");
	expect(comp).toHaveTextContent("Crazy");
	expect(comp).toHaveTextContent("By: user123");
});

it('mathces snapshot', function () {
	const mockInfo = {id: 5, name: "puzzle", creater: "rens"};

	const tree = renderer.create(<NonogramInfo info={mockInfo}></NonogramInfo>).toJSON();
	expect(tree).toMatchSnapshot();
});