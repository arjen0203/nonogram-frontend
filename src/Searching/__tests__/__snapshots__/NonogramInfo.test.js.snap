// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="nonoogram-info"
  data-testid="nonoInfo"
  onClick={[Function]}
>
  <div
    className="gram-info-name"
  >
    puzzle
  </div>
  <div
    className="gram-info-creater"
  >
    By: 
    rens
  </div>
</div>
`;
